### PR TITLE
addressed comments from @kqualters-elastic

### PR DESCRIPTION
**Pull request by mitodrummer**
Addressed a number of comments from Kevin. Mostly around use of eslint ignore on useEffect deps as well as all the function renderThis() stuff.

### CODE REVIEW

File: x-pack/plugins/session_view/public/components/process_tree_node/index.test.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ describe('ProcessTreeNode component', () => {
         <ProcessTreeNode isSessionLeader process={processMock} />
       );
 
-      expect(renderResult.container.textContent).toEqual(' bash started by  vagrant');
+      expect(renderResult.container.textContent).toEqual(' bash started by  vagrant');
```
Comment (mitodrummer) [inline x-pack/plugins/session_view/public/components/process_tree_node/index.test.tsx:39]: nbsp -> regular space

File: x-pack/plugins/session_view/public/components/process_tree_node/index.test.tsx [hunk 2]
```diff
@@ -45,17 +45,6 @@ describe('ProcessTreeNode component', () => {
     //   expect(renderResult.queryByText(/orphaned/i)).toBeTruthy();
     // });
 
-    it('renders user icon for user entered process', async () => {
```
Comment (mitodrummer) [inline x-pack/plugins/session_view/public/components/process_tree_node/index.test.tsx:48]: The user icon indicator is removed. We will probably not include any visual indication of user typed in v1 as it is an approximation.

File: x-pack/plugins/session_view/public/components/process_tree/index.tsx [hunk 3]
```diff
@@ -177,11 +169,11 @@ export const ProcessTree = ({
         ref={selectionAreaRef}
         css={styles.selectionArea}
       />
-      {hasNextPage &&
-        renderLoadMoreButton(
+      {hasNextPage && (
+        <EuiButton fullWidth onClick={fetchNextPage} isLoading={isFetching}>
           <FormattedMessage id="xpack.sessionView.loadNext" defaultMessage="Load next" />,
```
Comment (opauloh) [inline x-pack/plugins/session_view/public/components/process_tree/index.tsx]: I'll remove this comma also

File: x-pack/plugins/session_view/public/components/process_tree_alerts/index.tsx [hunk 4]
```diff
@@ -36,7 +36,7 @@ export function ProcessTreeAlerts({ alerts }: ProcessTreeAlertsDeps) {
     return http.basePath.prepend(`/app/security/rules/id/${alert.kibana?.alert.rule.uuid}`);
   };
 
-  const renderAlertDetails = (alert: ProcessEvent, index: number) => {
+  const renderedAlerts = alerts.map((alert: ProcessEvent, index: number) => {
```
Comment (opauloh) [inline x-pack/plugins/session_view/public/components/process_tree_alerts/index.tsx]: this should be wrapped in useMemo, I'll push a version with that

**Review by opauloh — APPROVED** [review]
lgtm

**Review by zizhouW — APPROVED** [review]

**Pull request merged**